File: solutions/clojure/two-fer/1/src/two_fer.clj
(ns two-fer)

;; The function two-fer replies with a sharing phrase, with name if provided, otherwise second-person-pronoun you.
(defn two-fer
      ([] ;; zero arity case, recur with second-person-pronoun
        (two-fer "you"))
      ([name] ;; single arity case, format with name provided
        (format "One for %s, one for me." name)))

File: solutions/r/two-fer/1/two-fer.R
# given input, responed with sharing phrase
two_fer <- function(input) {
  # if input is missing, respond with you, otherwise with name
  if (missing(input)) {
    print ("One for you, one for me.")
  } else {
    sprintf ("One for %s, one for me.", input)
  }
}
